internal/sshtunneler/tunneler.go

```diff
@@ -38,12 +38,12 @@ type State interface {
 
 // ControllerInfo defines an interface to fetch the controller's address.
 type ControllerInfo interface {
-	Addresses() network.SpaceAddresses
+	Addresses() (network.SpaceAddresses, error)
 }
 
 // SSHDialer defines an interface to establish an SSH connection over a provided connection.
 type SSHDial interface {
-	Dial(conn net.Conn, username string, privateKey gossh.Signer) (*gossh.Client, error)
+	Dial(conn net.Conn, username string, privateKey gossh.Signer, hostKeyCallback gossh.HostKeyCallback) (*gossh.Client, error)
```

Comment:
we could define 
`type DialFunction func(conn net.Conn, username string, privateKey gossh.Signer, hostKeyCallback gossh.HostKeyCallback) (*gossh.Client, error)`